### DISCUSSION TITLE
'Viewport' property for data-sveltekit-preload-data

### DISCUSSION THREAD

**Discussion by Devr-pro** [sveltejs/kit#9090] [General]
In SvelteKit preload data attributes are pretty awesome stuff, but I was wondering why there's no viewport like property for `data-sveltekit-preload-data` attribute? I guess, surely there might be reason behind this like any bad practices, but something like `data-sveltekit-preload-data="viewport"` would be so nice, like prefetching that page's data when that link or element comes in viewport and this would'nt be a bad practice too!
Please, enlighten me up if there's a cause behind this! Thanks.

**Comment by gurumaxi** [comment]
I was wondering the same, the viewport option would be so handy e.g. for a mobile web app that needs to feel very snappy. Right now, I guess the only option we have is to manually preload the data with `preloadData()`.

**Comment by andriytyurnikov** [comment]
Is it possible to prefetch data without parsing/evaluation? No DOM, no CSSOM.
